include/opendht/peer_discovery.h

```diff
@@ -0,0 +1,208 @@
+/*
+ *  Copyright (C) 2014-2019 Savoir-faire Linux Inc.
+ *  Author(s) : Mingrui Zhang <mingrui.zhang@savoirfairelinux.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "sockaddr.h"
+#include "infohash.h"
+
+#ifdef _WIN32
```

Comment:
move to cpp

Reply:
header required for OPENDHT_PUBLIC?

Reply:
`#include "def.h"` if you want `OPENDHT_PUBLIC`

Reply:
Also I think Adrien is talking about all the socket headers :)

Reply:
"def.h" is the only header necessary for `OPENDHT_PUBLIC`

Reply:
Got it ;~)